api/apps/services/dataset_api_service.py

```
@@ -1457,3 +1468,952 @@ async def search_datasets(tenant_id: str, req: dict):
+    frontend gets the authoritative numbers regardless of any
+    JSON-vs-column drift.
+    """
+    raw = row.get("content_with_weight") or ""
+    payload: dict = {}
+    if isinstance(raw, str) and raw.strip():
+        try:
+            parsed = json.loads(raw)
+            if isinstance(parsed, dict):
+                payload = parsed
+        except Exception:
+            pass
```

Ruff (0.15.20)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)